### PR TITLE
feat: Allow pregenerated ids

### PR DESCRIPTION
**Pull request by mkg20001**
Needed for https://github.com/ipfs/js-ipfsd-ctl/pull/284

### CODE REVIEW

File: src/core/components/init.js [hunk 1]
```diff
@@ -61,10 +61,20 @@ module.exports = function init (self) {
...
+        } else if (process.env.IPFS_PREGENERATED_PRIVATE_KEY) {
+          // Use pregenerated Id from env
+          self.log('using pregenerated id from env')
+          peerId.createFromPrivKey(Buffer.from(process.env.IPFS_PREGENERATED_PRIVATE_KEY, 'base64'), cb)
+        } else {
+          // Generate peer identity keypair + transform to desired format + add to config.
+          opts.log(`generating ${opts.bits}-bit RSA keypair...`, false)
+          self.log('generating peer id: %s bits', opts.bits)
+          peerId.create({ bits: opts.bits }, cb)
+        }
```
Comment (diasdavid) [inline src/core/components/init.js]: Instead of calling it "pregen ID", just say that it takes a Private Key.

Comment (diasdavid) [reply src/core/components/init.js]: And make it an option and not a env variable.

Comment (alanshaw) [reply src/core/components/init.js]: +1 for an option. i.e. a top level `privateKey` option to the constructor that you can also pass as a CLI option `--private-key=...`.

**Also needs to be documented in the README.**

**Comment by mkg20001** [comment]
@alanshaw Added test

**Comment by mkg20001** [comment]
Also the `--private-key` option would be needed in go-ipfs, too.

**Comment by Mr0grog** [comment]
I know I’m way late to this, but reading it, I worry that it won’t be clear that the private key is being used to generate a *peer ID* here (as opposed to, say, encrypting the repo contents on disk or something).

At the very least, I think it would be really helpful to state that this is the key for generating a peer ID in the README & CLI help. I’m kind of wondering if a different name for the option might be better, though, like `peerIdKey` or something  :\

**Comment by alanshaw** [comment]
@mkg20001 I have rebased and tweaked the README a little - hope that's ok :D

Lets get CI green and then get this merged!

**Comment by alanshaw** [comment]
@Mr0grog IMHO just `private-key` is ok here since it is the private key for your IPFS node.

**Pull request merged**